benchexec/tools/korn.py

```diff
@@ -0,0 +1,50 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+import benchexec.result as result
+import benchexec.tools.template
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for Korn, a software verifier based on Horn-clauses.
+    URL: https://github.com/gernst/korn
+    """
+
+    REQUIRED_PATHS = [
+        "run",
+        "korn.jar",
+        "z3",
+        "eld",
+        "eld.jar",
+    ]
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("run")
+
+    def version(self, executable):
+        return self._version_from_tool(executable)
+
+    def name(self):
+        return "Korn"
+
+    def determine_result(self, run):
+        """
+        This is literally the output from the underlying CHC solver
+        """
+
+        for line in run.output:
```

Comment:
This could be simplified, `strip()` is probably not necessary anymore except if there is whitespace before or after the result on the line. And you could even do `if "unsat" in run.output:`. But current code also works.

Reply:
right, but `"sat" in run.output` would neet to follow the test for `"unsat"` and that seems not very robust against changing the order of statements ;) the `strip()` was in fact necessary when I ran the experiments on Monday